RecoTracker/TkTrackingRegions/plugins/AreaSeededTrackingRegionsBuilder.cc

```diff
@@ -228,29 +234,93 @@ std::unique_ptr<TrackingRegion> AreaSeededTrackingRegionsBuilder::Builder::regio
+	// If we have objected used for seeding, loop over objects and find overlap with the found region. Return overlaps as tracking regions to use
+	for(const auto& object : *candidates.first) {
+		float dEta_Cand = candidates.second.first;
+		float dPhi_Cand = candidates.second.second;
+	       	float eta_Cand = object.eta();
+		float phi_Cand = object.phi();
+          	float dEta_Cand_Point = std::abs(eta_Cand-meanEta);
+          	float dPhi_Cand_Point = std::abs(deltaPhi(phi_Cand,meanPhi));
+
+		if(dEta_Cand_Point > (dEta_Cand + dEta) || dPhi_Cand_Point > (dPhi_Cand + dPhi)) continue;
+
+		float  etaMin_RoI = std::max(eta_Cand-dEta_Cand,meanEta-dEta);
+		float  etaMax_RoI = std::min(eta_Cand+dEta_Cand,meanEta+dEta);
+
+		float  phi_Cand_minus  = normalizedPhi(phi_Cand-dPhi_Cand);
+		float  phi_Point_minus = normalizedPhi(meanPhi-dPhi);
+		float  phi_Cand_plus  = normalizedPhi(phi_Cand+dPhi_Cand);
+		float  phi_Point_plus = normalizedPhi(meanPhi+dPhi);
+
+		float phiMin_RoI = deltaPhi(phi_Cand_minus,phi_Point_minus)>0. ? phi_Cand_minus : phi_Point_minus ;	
+		float phiMax_RoI = deltaPhi(phi_Cand_plus,phi_Point_plus)<0. ? phi_Cand_plus : phi_Point_plus;
+
+
+		const auto meanEtaTemp = (etaMin_RoI+etaMax_RoI)/2.f;
+		auto meanPhiTemp = (phiMin_RoI+phiMax_RoI)/2.f;
+		if( phiMax_RoI < phiMin_RoI ) meanPhiTemp-=M_PI;
+	        meanPhiTemp = normalizedPhi(meanPhiTemp);
+	        
+		const auto dPhiTemp = deltaPhi(phiMax_RoI,meanPhiTemp);
+		const auto dEtaTemp = etaMax_RoI-meanEtaTemp;
+
+		const auto x = std::cos(meanPhiTemp);
+  	  	const auto y = std::sin(meanPhiTemp);
+  	  	const auto z = (x*x+y*y)/std::tan(2.f*std::atan(std::exp(-meanEtaTemp))); // simplify?
+
+  		LogTrace("AreaSeededTrackingRegionsBuilder") << "Direction x,y,z " << x << "," << y << "," << z
+                                               << " eta,phi " << meanEtaTemp << "," << meanPhiTemp
+                                               << " window eta " << (meanEtaTemp-dEtaTemp) << "," << (meanEtaTemp+dEtaTemp)
+                                               << " phi " << (meanPhiTemp-dPhiTemp) << "," << (meanPhiTemp+dPhiTemp);
+
+  		return std::make_unique<RectangularEtaPhiTrackingRegion>(
+      			GlobalVector(x,y,z),
+      			origin.first, // GlobalPoint
+      			m_conf->m_ptMin,
+      			m_conf->m_originRadius,
+      			origin.second,
+      			dEtaTemp,
+      			dPhiTemp,
+      			m_conf->m_whereToUseMeasurementTracker,
+      			m_conf->m_precise,
+      			m_measurementTracker,
+      			m_conf->m_searchOpt
+  		);  
+	}
+	// Have to retun nullptr here to ensure that we always return something
+	return nullptr;
 
-  const auto x = std::cos(meanPhi);
-  const auto y = std::sin(meanPhi);
-  const auto z = (x*x+y*y)/std::tan(2.f*std::atan(std::exp(-meanEta))); // simplify?
+  }
+  else{
+  	const auto x = std::cos(meanPhi);
+  	const auto y = std::sin(meanPhi);
+  	const auto z = (x*x+y*y)/std::tan(2.f*std::atan(std::exp(-meanEta))); // simplify?
```

Comment:
(Same as above)

```(x*x+y*y)``` is identically equal to 1 here: since you want to reduce the cpu cosumption in your algo, please avoid useless computations

Moreover, and in the same spirit: ```(x,y,z) ``` are only computed to be passed to the constructor of the GlobalVector a few lines below: did you check wheter by using the constructor with ```(eta, phi, r)```, that can avoid making those computations yourself, you can further optimize?

Reply:
oh yes, simplify and eventually even speed it up using reduced precision....